salt/states/mongodb.py

```diff
@@ -0,0 +1,433 @@
+# -*- coding: utf-8 -*-
+'''
+Management of Mongodb users and databases
+=========================================
+
+.. note::
+    This module requires PyMongo to be installed.
+'''
+
+# Define the module's virtual name
+__virtualname__ = 'mongodb'
+
+
+def __virtual__():
+    for i in ('mongodb.user_exists', 'mongodb.db_exists',):
```

Comment:
Either the mongodb module should be loaded and all its funcs will be available or it won't be. There shouldn't be a need to do two full loops here. One should suffice.